### PR TITLE
retry if error 11(EAGAIN) is encountered

### PR DESCRIPTION
**Pull request by changweige**
EAGAIN is a common error code that caller should handle, usually by retrying

Signed-off-by: Changwei Ge <gechangwei@bytedance.com>

## Reason for This PR

`[Author TODO: add issue # or explain reasoning.]`

## Description of Changes

`[Author TODO: add description of changes.]`

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

## PR Checklist

`[Author TODO: Meet these criteria.]`
`[Reviewer TODO: Verify that these criteria are met. Request changes if not]`

- [x] All commits in this PR are signed (`git commit -s`).
- [x] The reason for this PR is clearly provided (issue no. or explanation).
- [x] The description of changes is clear and encompassing.
- [x] Any required documentation changes (code and docs) are included in this PR.
- [x] Any newly added `unsafe` code is properly documented.
- [x] Any user-facing changes are mentioned in `CHANGELOG.md`.

### CODE REVIEW

File: src/server.rs [hunk 1]
```diff
@@ -122,6 +124,12 @@ impl<T: Read + Write + ScmSocket> ClientConnection<T> {
                     return Ok(vec![]);
                 }
                 Err(ConnectionError::StreamReadError(inner)) => {
+                    #[cfg(target_os = "linux")]
+                    if inner.errno() == libc::EAGAIN {
+                        sleep(Duration::from_micros(200));
```
Comment (jiangliu) [inline src/server.rs]: 200ms seems a little long, how about 10?

Comment (changweige) [reply src/server.rs]: It's microseconds(us) rather milliseconds (ms)

Comment (changweige) [reply src/server.rs]: Do you mean 10us?

Comment (changweige) [reply src/server.rs]: Changed to 20us

**Comment by changweige** [comment]
reopen to trigger all checks again

**Pull request merged**